test/functional/test_framework/test_node.py

```diff
@@ -433,8 +433,8 @@ def batch(self, requests):
 
     def send_cli(self, command=None, *args, **kwargs):
         """Run bitcoin-cli command. Deserializes returned string as python object."""
-        pos_args = [str(arg).lower() if type(arg) is bool else str(arg) for arg in args]
-        named_args = [str(key) + "=" + str(value) for (key, value) in kwargs.items()]
+        pos_args = [str(arg).lower() if type(arg) is bool else (json.dumps(arg) if (isinstance(arg, dict) or isinstance(arg, list)) else str(arg)) for arg in args]
+        named_args = [str(key) + "=" + (json.dumps(value) if (isinstance(value, dict) or isinstance(value, list)) else str(value)) for (key, value) in kwargs.items()]
```

Comment:
paging @jnewbery, looks ok?

Reply:
Definitely agree with pulling this monstrosity out into a function :)

bools in named args were fixed here: https://github.com/bitcoin/bitcoin/pull/14938#discussion_r243390546 but that PR wasn't merged.

Reply:
I've pulled this into a new function which is easier to read. It also fixes the bools in named args issue.

Reply:
Since this issue affects multiple PRs, I've split it into a separate PR: #15301